### ISSUE TITLE
Issue when buying certain weapons, ammo, and equipping certain weapons.

### ISSUE THREAD

**Issue by hhaslam11** [hhaslam11/Text-Fighter#70]
"You already own this weapon."
:-1:

**Comment by hhaslam11** [comment]
Update: when trying to purchase ammo, it attempts to buy a sniper rifle instead.

**Comment by hhaslam11** [comment]
When trying to buy POWER, a rifle is attempted to be purchased. Everything else seems fine. Indexing get's confused after sniper. But why.....

**Comment by BrendonButler** [comment]
@hhaslam11 math is hard :/

**Comment by hhaslam11** [comment]
Especially when it's not even my code.. (@htn17)

**Comment by hhaslam11** [comment]
Btw, @htn17 (If you're following the thread right now), what does 'j' stand for in:

```
int[] offset = new int[arrayWeapon.size()];
            for(int i = 0; i < arrayWeapon.size(); i++){
               if(arrayWeapon.get(i).owns()){
                   Ui.println((j + 1) + ") " + arrayWeapon.get(i).getName());
                   offset[j] = i - j;
                   j++;
               }
            }
```

(Or what role does it play)

**Comment by hhaslam11** [comment]
or this

```
for(int i = 0; i < Weapon.arrayWeapon.size(); i++){
                if(Weapon.arrayWeapon.get(i).isBuyable()){
                    Ui.println((j + 1) + ") " + Weapon.arrayWeapon.get(i).getName());
                    Ui.println("   Price: " + Weapon.arrayWeapon.get(i).price);
                    Ui.println("   Level: " + Weapon.arrayWeapon.get(i).level);
                    weaponShopOffset[j] = i - j;
                    j++;
                    Ui.println();
                }
            }
```

**Comment by hhaslam11** [comment]
Looks like it's just simply not taking in account the POWER and Ammo. Only counting the weapons in the array. Now to patch that.
(Updating myself in a one-way conversation ;))

**Comment by hhaslam11** [comment]
I thought I fixed it by making a copy of arrayWeapon(), containing only buyable weapons (was easier than whatever was being done before. Got too confused so I just rewrote most of it). However, now that I think about it, buy() is being called on the copy of the array, not the array that the rest of the program uses. This might cause the weapon to not actually be bought.

Also, the buyammo menu is messed up (Can't exit & probably a bunch of other bugs.)
I'll continue working on these issues tomorrow.
